apps/andi/lib/andi_web/live/edit_live_view.ex

```diff
@@ -2,89 +2,171 @@ defmodule AndiWeb.EditLiveView do
+            </div>
+            <div class="metadata-form__homepage">
+              <%= Form.label(fp, :homepage, "Data Homepage URL", class: "label") %>
+              <%= Form.text_input(fp, :homepage, class: "input") %>
+            </div>
+          <% end %>
+          <%= Form.inputs_for f, :technical, fn fp -> %>
+            <div class="metadata-form__format">
+              <%= Form.label(fp, :sourceFormat, "Format", class: "label label--required") %>
+              <%= Form.text_input(fp, :sourceFormat, [class: "input--text input", readonly: true]) %>
+              <%= error_tag(fp, :sourceFormat) %>
+            </div>
+            <div class="metadata-form__level-of-access">
+              <%= Form.label(fp, :private, "Level of Access", class: "label label--required") %>
+              <%= Form.select(fp, :private, get_level_of_access_options(), class: "select") %>
+              <%= error_tag(fp, :private) %>
+            </div>
+          <% end %>
+        <div class="metadata-form__cancel-btn">
+          <%= Link.button("Cancel", to: "/", method: "get", class: "btn btn--cancel") %>
+        </div>
+        <div class="metadata-form__save-btn">
+          <%= Link.button("Next", to: "#", id: "next-button", class: "btn btn--next") %>
+          <%= Form.submit("Save", id: "save-button", class: "btn btn--save") %>
+        </div>
+      </div>
+      <div>
+        <%= if @is_saved do %>
+          <div id="success-message" class="metadata__success-message">Saved Successfully</div>
+        <% end %>
+      </div>
+
     """
   end
 
   def mount(%{dataset: dataset}, socket) do
-    {:ok, assign(socket, dataset: dataset)}
+    {:ok,
+     assign(socket, id: dataset.id, dataset: dataset, changeset: Andi.DatasetSchema.changeset(dataset), is_saved: false)}
   end
 
-  defp get_private(%{technical: %{private: true}}), do: "Private"
-  defp get_private(_), do: "Public"
+  def handle_event(
+        "validate",
+        %{"dataset_schema" => dataset_schema} = event,
+        socket
+      ) do
+    {:noreply, assign(socket, changeset: apply_changes(dataset_schema), is_saved: false)}
+  end
+
+  def handle_event("save", %{"dataset_schema" => dataset_schema} = event, socket) do
+    change = apply_changes(dataset_schema)
+
+    case change.valid? do
+      true ->
+        schema = Ecto.Changeset.apply_changes(change)
+        original_dataset = socket.assigns.dataset
+
+        %{
+          original_dataset
+          | business: Map.merge(Map.from_struct(original_dataset.business), Map.from_struct(schema.business)),
+            technical: Map.merge(Map.from_struct(original_dataset.technical), Map.from_struct(schema.technical))
+        }
+        |> SmartCity.Dataset.new()
+        |> send_dataset_update
+
+        {:noreply, assign(socket, changeset: change, is_saved: true, update_stepper_state: "meta-data-save")}
```

Comment:
What's the update stepper state?

Reply:
The arbitrary property that we have to add to get live view to re-render the view if there is no diff... At some point, I'm assuming we well perform some sort of UI action on save